### PR TITLE
Adding documentation on template functions

### PR DESCRIPTION
**Pull request by mattfarina**
This includes documentation of most of the template functions
available within Helm templates.

### CODE REVIEW

File: content/en/docs/chart_template_guide/function_list.md [hunk 1]
```diff
@@ -0,0 +1,1951 @@
...
+### now
+
+The current date/time. Use this in conjunction with other date functions.
+
+### ago
+
+The `ago` function returns duration from time.Now in seconds resolution.
+
+```
+ago .CreatedAt"
```
Comment (technosophos) [inline content/en/docs/chart_template_guide/function_list.md:977]: Extra quotation mark here

**Comment by mattfarina** [comment]
Good catch on the `tpl` function. I found a few more I missed. I'll do that as a follow-up PR.

File: content/en/docs/chart_template_guide/function_list.md (hunk 1 above)
Comment (mattfarina) [reply content/en/docs/chart_template_guide/function_list.md:977]: Fixed. Thanks

File: content/en/docs/chart_template_guide/function_list.md [hunk 2]
```diff
@@ -0,0 +1,1951 @@
...
+The `pluck` function makes it possible to give one key and multiple maps, and
+get a list of all of the matches:
+
+```
+pluck "name1" $myDict $myOtherDict
+```
+
+The above will return a `list` containing every found value (`[value1 otherValue1]`).
+
+If the give key is _not found_ in a map, that map will not have an item in the
```
Comment (glensc) [inline content/en/docs/chart_template_guide/function_list.md:1175]: > If the give**n** key is _not found_ in a map

**Comment by glensc** [comment]
The outcome is at this address: https://helm.sh/docs/chart_template_guide/function_list/

**Comment by glensc** [comment]
@mattfarina I noticed that not all headings are listed in the toc beginning of the page:

- Kubernetes and Chart Functions
- File Functions

**Pull request merged**